collectors/hra/collector.py

```diff
@@ -0,0 +1,137 @@
+                data['application_full_title'] = application['ApplicationFullTitle']
+                data['isrctn'] = application['ISRCTN']
+                data['nct'] = application['NCT']
+                data['additional_ref_numbers'] = application['AdditionalReferenceNumbers']
+                data['duration_of_study_in_uk'] = application['DurationOfStudyInUK']
+                data['research_summary'] = application['ResearchSummary']
+                data['eudra_ct'] = application['EudraCT']
+                data['social_value'] = application['SocialValue']
+                data['recuitment_arrangements'] = application['RecruitmentArrangements']
+                data['risk_and_benefit'] = application['RiskAndBenefit']
+                data['participants_protection_and_care'] = application['ParticipantsProtectionAndCare']
+                data['informed_consent'] = application['InformedConsent']
+                data['applicant_and_staff_suitability'] = application['ApplicantAndStaffSuitability']
+                data['independent_review'] = application['IndependentReview']
+                data['supporting_info_suitability'] = application['SupportingInfoSuitability']
+                data['other_comments'] = application['OtherComments']
+                data['research_summary_suitability'] = application['ResearchSummarySuitability']
+
+                # Insert into database only if app_id not inserted already
+                check_id = conn['warehouse'].query("select count(*) from hra where application_id='%s'" % str(application['ApplicationID']))
+                for row in check_id:
+                    if row[0]['count'] == 0:
+                        record = Record.create(endpoint, data)
+                        base.writers.write_record(conn, record)
+
+                success += 1
+                print("Successful parsing iteration")
+            except Exception as exception:
+
+                # Log warning
+                errors += 1
+                logger.warning('Collecting error: %s', repr(exception))
+                print(exception)
+
+    def get_to_date():
+        to_date = conn['warehouse'].query('select min(api_date_from) from hra').result_proxy.first()[0] - timedelta(days=1)
+        return to_date
+
+    try:
+        table = conn['warehouse'].load_table('hra')
+        if table.count() == 0:
+            to_date = date.today()
+        else:
+            to_date = get_to_date()
+
+    except sqlalchemy.exc.NoSuchTableError:
+        to_date = date.today()
+
+    query_period = 6
+    from_date = to_date - timedelta(days=query_period)
+    response, endpoint = make_request(from_=from_date, to=to_date, url=url, filter=None)
+    parse_response(response, endpoint, from_date, to_date, errors, success)
+
+    if (len(response.json()) > 0):
+        while True:
+            to_date = get_to_date()
+            from_date = to_date - timedelta(days=query_period)
+            print('Sleeping for 30sec.')
```

Comment:
I mean if you prefer to have separate function for it (we could write this `data['']=` just inside our cycle without function) than is should be something like:

> parser.py

```
def parse_records(response):
   records = []
   # fill records
   return records
```

and then in `collect` write it.